pkg-r/vignettes/get-started.Rmd

```diff
@@ -0,0 +1,347 @@
+```
+
+Congrats, you now have a chat interface powered by an LLM of your choice! 🎉
+
+```{r, fig.cap="Screenshot of a conversation using shinychat.", fig.align='center', out.extra='class="rounded shadow"'}
+knitr::include_graphics("images/chat-quick-start.png")
+```
+
+### Inspect the code
+
+Let's take a closer look at the code in `app.R`. 
+
+```r
+library(shiny)
+library(shinychat)
+
+ui <- bslib::page_fluid(
+  # Add a chat UI element
+  chat_ui("chat")
+)
+
+server <- function(input, output, session) {
+  # Initialize a chat with your chosen model provider
+  chat <- ellmer::chat_openai(system_prompt = "You are a helpful assistant.")
+  
+  # Listen for user input and communicate with the model
+  observeEvent(input$chat_user_input, {
+    stream <- chat$stream_async(input$chat_user_input)
+    chat_append("chat", stream)
+  })
+}
+
+shinyApp(ui, server)
+```
+
+A shinychat chatbot includes three core steps:
+
+1. **Create a chat UI element** with [`chat_ui()`](https://posit-dev.github.io/shinychat/reference/chat_ui.html).
+
+2. **Initialize a chat** with a `chat_*()` function, like `chat_openai()`, in the server function. You can use the `system_prompt` argument to supply a [system prompt](https://ellmer.tidyverse.org/articles/prompt-design.html).
+
+3. **Set up a reactive listener** with `observeEvent()` that waits for the user to submit a message (`input$chat_user_input`). When a message is received:
+
+    * Send the input to the LLM using `chat$stream_async()`, which returns asynchronously streaming results from the LLM. This means the results will appear in chunks, so the user doesn’t have to wait for the full response.
+    * Append the response to the `chat_ui()` element with `chat_append()`, so the user can see the model’s reply appear live as it's generated.
+
+### Switch the model
+
+To switch the model provider, use a different `chat_*()` function.
+
+```r
+library(shiny)
+library(shinychat)
+
+ui <- bslib::page_fluid(
+  chat_ui("chat")
+)
+
+server <- function(input, output, session) {
+  # Switch this chat_*() function to use a different model provider
+  chat <- ellmer::chat_ollama(system_prompt = "You are a helpful assistant.")
+  
+  observeEvent(input$chat_user_input, {
+    stream <- chat$stream_async(input$chat_user_input)
+    chat_append("chat", stream)
+  })
+}
+
+shinyApp(ui, server)
+```
+
+### Add a system prompt
+
+Use the `chat_*()` function's `system_prompt` argument to provide the LLM with more information about how you would like it to behave. 
+
+```r
+chat <- ellmer::chat_ollama(system_prompt = "You are a helpful assistant")
+```
+
+To learn more about writing system prompts, see ellmer's [Prompt design](https://ellmer.tidyverse.org/articles/prompt-design.html) vignette. Generally, we recommend writing the system prompt in a separate markdown file, but if your prompt is short you can also supply it directly as a string to the `system_prompt` argument.
+
+### Add messages and suggestions on startup
```

Comment:
This gives the impression that suggestions can only be done at startup -- it'd be great to have some instruction on how to get suggestions in responses, like this section does https://shiny.posit.co/py/docs/genai-chatbots.html#suggest-input